tests/invalid_models_tests/test_relative_fields.py

```diff
@@ -200,6 +211,16 @@ class ModelM2M(models.Model):
                 id='fields.W341',
             )
         )
+        expected.append(
+            DjangoWarning(
+                'limit_choices_to has no effect on ManyToManyField'
+                'with an explicit intermediary model specified using'
+                'the through parameter',
+                hint=None,
+                obj=field,
+                id='fields.W343'
+            )
+        )
```

Comment:
While you're at it could you initialize the `expected` list with all the warnings once instead of calling `append()` twice.